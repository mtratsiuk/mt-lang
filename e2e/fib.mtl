(def fib
  (|n|
    (cond
      ((<= n 1)
        n)
      (else
        (+
          (fib (- n 2))
          (fib (- n 1)))))))

(print (fib 13))
