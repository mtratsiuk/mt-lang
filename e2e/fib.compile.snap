const fib = ((n) => {
  return (() => {
    if ((n <= 1)) {
      return n;
    } else {
      return (fib((n - 2)) + fib((n - 1)));
    }
  })();
});

console.log(String(fib(13)));
